src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -304,100 +262,156 @@ template <class T>
 Status
 ledgerFromRequest(
     T& ledger,
-    GRPCContext<rpc::v1::GetAccountInfoRequest>& context)
+    GRPCContext<org::xrpl::rpc::v1::GetAccountInfoRequest>& context)
 {
-    static auto const minSequenceGap = 10;
-
     ledger.reset();
 
-    rpc::v1::GetAccountInfoRequest& request = context.params;
-    auto& ledgerMaster = context.ledgerMaster;
+    org::xrpl::rpc::v1::GetAccountInfoRequest& request = context.params;
 
-    using LedgerCase = rpc::v1::LedgerSpecifier::LedgerCase;
+    using LedgerCase = org::xrpl::rpc::v1::LedgerSpecifier::LedgerCase;
     LedgerCase ledgerCase = request.ledger().ledger_case();
 
     if (ledgerCase == LedgerCase::kHash)
     {
         uint256 ledgerHash = uint256::fromVoid(request.ledger().hash().data());
-        if (ledgerHash.size() != request.ledger().hash().size())
-            return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
-
-        ledger = ledgerMaster.getLedgerByHash(ledgerHash);
-        if (ledger == nullptr)
-            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        return getLedger(ledger, ledgerHash, context);
     }
     else if (ledgerCase == LedgerCase::kSequence)
     {
-        ledger = ledgerMaster.getLedgerBySeq(request.ledger().sequence());
-
-        if (ledger == nullptr)
+        return getLedger(ledger, request.ledger().sequence(), context);
+    }
+    else if (
+        ledgerCase == LedgerCase::kShortcut ||
+        ledgerCase == LedgerCase::LEDGER_NOT_SET)
+    {
+        auto const shortcut = request.ledger().shortcut();
```

Comment:
seems like this would read better as a switch statement

Reply:
will change to switch